### PR TITLE
[algo] Adding CISPO policy loss

### PR DESCRIPTION
**Pull request by twkillian**
### What does this PR do?

This PR adds CISPO to `core_algos.py` to start integrations toward a full implementation of scaleRL.

### Checklist Before Starting

- [X] Search for similar PRs. Paste at least one query link here: ...
- [X] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### API and Usage Example

CISPO is a sampled policy gradient loss that adopts a lot from the REINFORCE family of algorithms. How it differs from GRPO, etc is that the IS ratio is clipped directly rather than the fully policy clipping that is done in PPO derivatives. This introduces two new hyperparameters `cispo_clip_ratio_high` and `cispo_clip_ratio_low` to handle this clipping. They are each defaulted to 0.2. 

Also, we've introduced a new policy loss function for CISPO, which is employed when adjusting the `loss_mode` in the run configuration. Altogether this looks like:

```python
actor_rollout_ref.actor.policy_loss.loss_mode=cispo \
actor_rollout_ref.actor.policy_loss.cispo_clip_ratio_high=0.2 \
actor_rollout_ref.actor.policy_loss.cispo_clip_ratio_low=0.2 \
```

### CODE REVIEW

File: verl/utils/reward_score/naive_dapo.py [hunk 1]
```diff
@@ -425,8 +425,8 @@ def grade_answer(given_answer: str, ground_truth: str) -> tuple[bool, str]:
             elif _str_is_int(ground_truth_elem) != _str_is_int(given_elem):
                 # if the ground truth answer is an integer, we require the given answer to be a strict match (no sympy.simplify)
                 is_correct = False
-            else:
-                is_correct = are_equal_under_sympy(ground_truth_elem, given_elem)
+            # else:
+            #     is_correct = are_equal_under_sympy(ground_truth_elem, given_elem)
```
Comment (nightlessbaron) [inline verl/utils/reward_score/naive_dapo.py]: do we want to remove this?

File: verl/utils/reward_score/prime_math/__init__.py [hunk 2]
```diff
@@ -55,7 +55,7 @@ def _parse_latex(expr: str) -> str:
     expr = expr.replace("\\tfrac", "\\frac")
     expr = expr.replace("\\dfrac", "\\frac")
     expr = expr.replace("\\frac", " \\frac")  # Play nice with mixed numbers.
-    expr = latex2text.LatexNodes2Text().latex_to_text(expr)
+    # expr = latex2text.LatexNodes2Text().latex_to_text(expr)
```
Comment (nightlessbaron) [inline verl/utils/reward_score/prime_math/__init__.py]: please also add the following comment # Added by Reasoning360

File: recipe/dapo/dapo_ray_trainer.py
Comment (nightlessbaron) [inline recipe/dapo/dapo_ray_trainer.py:1]: imo, it would be ideal to create another directory called cispo instead of adding modifications in dapo
